pkg/utils/grpc.go

```diff
@@ -0,0 +1,41 @@
+package utils
+
+import (
+	"context"
+	"fmt"
+
+	gateway "github.com/cs3org/go-cs3apis/cs3/gateway/v1beta1"
+	user "github.com/cs3org/go-cs3apis/cs3/identity/user/v1beta1"
+	rpc "github.com/cs3org/go-cs3apis/cs3/rpc/v1beta1"
+	revactx "github.com/cs3org/reva/v2/pkg/ctx"
+	"google.golang.org/grpc/metadata"
+)
+
+// Impersonate returns an authenticated reva context and the user it represents
+func Impersonate(userID *user.UserId, gwc gateway.GatewayAPIClient, machineAuthAPIKey string) (context.Context, *user.User, error) {
```

Comment:
Nice util. Is this func currently used somewhere?
And because we have a sunny day, i would love to see a unit test for this func 😄

Reply:
I'm using it in the ocis pendant of this PR.

Not sure what I am supposed to unit test here. It does requests to `GetUser` and `Authenticate`. If I mock them away there is nothing left to be tested...

Reply:
hmm, i see.  lets merge it.